PPAK_CVS/Sources/Scenes/Product/ProductViewReactor.swift

```diff
@@ -65,16 +71,32 @@ final class ProductViewReactor: Reactor {
     switch mutation {
     case .updateProduct(let model):
       newState.model = model
+      
     case .goToHomeVC:
       newState.isPopProductVC = true
+      
     case .changeBookmarkState(let isBookmark):
-      newState.isBookmark = !isBookmark
+      self.updateBookmarkState(isBookmark: isBookmark)
+      newState.isBookmark = isBookmark
+      
     case let .showShareWindow(isShareButtonTapped):
       newState.isShareButtonTapped = isShareButtonTapped
+      
     case let .setItem(newImage):
       newState.shareImage = newImage
     }
 
     return newState
   }
+  
+  /// 북마크 상태를 UserDefaults에 적용하는 메서드입니다.
+  private func updateBookmarkState(isBookmark: Bool) {
+    if isBookmark {
+      if ProductStorage.shared.contains(currentState.model) == false {
+        ProductStorage.shared.add(currentState.model)
+      }
+    } else {
+      ProductStorage.shared.remove(currentState.model)
+    }
+  }
```

Comment:
depth는 줄일 수 있다면 줄여보는 것도 좋을 것 같아요.

```suggestion
    // guard isBookmark else { return } // 또는
    if !isBookmark { return }
    
    let storage = ProductStorage.shared
    
    if storage.contains(currentState.model) {
      storage.remove(currentState.model)
    } else {
      storage.add(currentState.model)
    }
```

Reply:
> depth는 줄일 수 있다면 줄여보는 것도 좋을 것 같아요.

아! 그렇네요 감사합니다!